### PR TITLE
Enhance contribution and use without docker

### PR DESCRIPTION
**Pull request by s-pace**
Introducing the use of `pipenv` to ease the way to use our tool outside Docker

cc #403

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,16 +20,39 @@ If you are Algolia employee and want to manage a DocSearch account,
...
+HELP_SCOUT_API_KEY=
+NB_HITS_MAX=
+
 WEBSITE_USERNAME=
 WEBSITE_PASSWORD=
 SLACK_HOOK=
 SCHEDULER_USERNAME=
 SCHEDULER_PASSWORD=
+MONITORING_API_KEY=
+PUBLIC_CONFIG_FOLDER=<path to your docsearch-scrapy directory> + /configs/public
```
Comment (pixelastic) [inline CONTRIBUTING.md]: Shouldn't this be:
```suggestion
PUBLIC_CONFIG_FOLDER=<path to your docsearch-scraper directory> + /configs/public
```

Comment (pixelastic) [reply CONTRIBUTING.md]: Also, if it's always a subdirectory of the scraper, we can guess the filepath easily, without the need for a variable, right?

File: cli/oh-my-zsh/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ## How to install
 
-- copy the ```docsearch``` folder in ```~/.oh-my-zsh/custom/plugins/```
-- enable the plugin in your ```zshrc``` by modify the plugins variable: ``plugins=([...], docsearch)
\ No newline at end of file
+- copy the `docsearch` folder in `~/.oh-my-zsh/custom/plugins/`
```
Comment (pixelastic) [inline cli/oh-my-zsh/README.md]: You should add a paragraph about what this plugin does. I didn't even know we had one, and I still don't really know what it does. Is that autocompletion?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,14 @@
 Hello you,
 
+## Install the scrapper without Docker
+
+We do recommend the use of [pipenv][1] to install the whole python environment
+
+- [Install pipenv][2]
+- `pipenv --python 2.7 install`
```
Comment (pixelastic) [inline CONTRIBUTING.md]:
```suggestion
- `pipenv install`
```

This should be enough, the Python version is already defined in the `Pipfile`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+
```
Comment (pixelastic) [inline CONTRIBUTING.md]: And  `pipenv shell` to drop into a shell with all the required dependencies, or `pipenv run ./docsearch` to start docsearch

**Comment by pixelastic** [comment]
And you should also remove `requirements.txt` now that we have `Pipfile`, it's no longer needed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (s-pace) [reply CONTRIBUTING.md]: Yes

File: cli/oh-my-zsh/README.md (hunk 2 above)
Comment (s-pace) [reply cli/oh-my-zsh/README.md]: I do not recall cc @maxiloc

Comment (pixelastic) [reply cli/oh-my-zsh/README.md]: If it's not used, let's remove it. But if it's useful, let's use it :)

**Comment by pixelastic** [comment]
Ah wait, you need to update the Travis scripts before merging. They use `requirements.txt`. They now need to use pipenv

**Comment by s-pace** [comment]
let's keep requirement.txt then. It is pretty convenient for both usage.

**Comment by s-pace** [comment]
@pixelastic makes sense. Please update Travis then. I do not know how to do it.

🙏

**Comment by pixelastic** [comment]
Ok, I'll have a look.

**Comment by pixelastic** [comment]
@s-pace @maxiloc Could you re-explain how DocSearch works with the different python versions?

I see that the new Pipfile uses python 2.7, and I know I can only run `./docsearch` personally when I'm using python 2.7. But I see that the Travis build sets a `USE_PYTHON3` to true and the subsequent docker image create uses python 3 for the tests. I also see we have a bunch of compatibility layers (`future`, `six`).

Why do we have this mix of python 2 and 3, and what is the recommended version for running DocSearch? My understanding was that DocSearch uses mostly python 2, but we would like to move to python 3 at some point, but the code would need a big rehaul to do so. Could you clarify the current state? 

Thanks!

**Comment by s-pace** [comment]
The scraper only works with python 2.7. I do not know why this env is set

**Comment by s-pace** [comment]
Closing it in favor of #411

**Pull request closed without merging**